rita/src/rita_common/port_emissary/mod.rs

```diff
@@ -3,22 +3,35 @@ use std::io::prelude::*;
 use std::u16;
 
 /// Helper function for parsing out port number from local_address column
-fn take_local_port(s: &str) -> u16 {
+fn take_local_port(s: &str) -> Result<u16, &'static str> {
     // second column in table contains local_address
-    let local_addr = s.split_whitespace().nth(1).unwrap();
+    let local_addr = match s.split_whitespace().nth(1) {
+        Some(addr) => addr,
+        None => return Err("Error parsing local_address column!"),
+    };
     // having a format like "00000000:14E9"
-    let port = local_addr.split(":").nth(1).unwrap();
-
-    u16::from_str_radix(port, 16).unwrap()
+    let port = match local_addr.split(":").nth(1) {
+        Some(port) => port,
+        None => return Err("Error parsing local_address column!"),
+    };
+
+    match u16::from_str_radix(port, 16) {
+        Ok(port) => Ok(port),
+        Err(err) => return Err("Error parsing port from local_address column!"),
+    }
 }
 
 /// Returns list of ports in use as seen in the UDP socket table (/proc/net/udp)
-fn used_ports() -> Vec<u16> {
-    let mut f = File::open("/proc/net/udp").expect("UDP socket table not found!");
+fn used_ports() -> Result<Vec<u16>, &'static str> {
+    let mut f = match File::open("/proc/net/udp") {
+        Ok(file) => file,
+        Err(err) => return Err("UDP socket table not found!"),
```

Comment:
you could return err and pass through the error? also the err must be giving you an unused warning right now.